Tests/Brokerages/Oanda/OandaBrokerageTests.cs

```diff
@@ -0,0 +1,116 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Newtonsoft.Json;
+using NUnit.Framework;
+using QuantConnect.Brokerages.Oanda;
+using QuantConnect.Brokerages.Oanda.DataType;
+using QuantConnect.Brokerages.Oanda.DataType.Communications;
+using QuantConnect.Brokerages.Oanda.Framework;
+using QuantConnect.Configuration;
+using QuantConnect.Interfaces;
+using QuantConnect.Securities;
+using Environment = QuantConnect.Brokerages.Oanda.Environment;
+
+namespace QuantConnect.Tests.Brokerages.Oanda
+{
+    [TestFixture]
+    public class OandaBrokerageTests : BrokerageTests
+    {
+        /// <summary>
+        ///     Creates the brokerage under test and connects it
+        /// </summary>
+        /// <returns>A connected brokerage instance</returns>
+        protected override IBrokerage CreateBrokerage(IOrderProvider orderProvider, IHoldingsProvider holdingsProvider)
+        {
+            var oandaBrokerage = new OandaBrokerage(orderProvider, holdingsProvider, 0);
+            var tokens = OandaBrokerageFactory.GetTokens();
+            var environment = Config.Get("oanda-environment");
+            if (environment == "sandbox")
+            { 
+                var requestString = EndpointResolver.ResolveEndpoint(Environment.Sandbox, Server.Account) + "accounts";
+                var accountResponse = oandaBrokerage.MakeRequest<AccountResponse>(requestString, "POST");
+                oandaBrokerage.SetAccountId(accountResponse.accountId);
+                oandaBrokerage.SetEnvironment("sandbox");
+                
+            }
+            else
+            {
+                oandaBrokerage.SetAccountId(Convert.ToInt32(Config.Get("oanda-account-id")));
+                oandaBrokerage.SetEnvironment(Config.Get("oanda-environment"));    
+            }
+
+            var qcUserId = OandaBrokerageFactory.Configuration.QuantConnectUserId;
+            oandaBrokerage.SetTokens(qcUserId, tokens.AccessToken, tokens.RefreshToken, tokens.IssuedAt,
+                TimeSpan.FromSeconds(tokens.ExpiresIn));
+
+            oandaBrokerage.InitializeInstrumentSecurityTypeMap();
+
+            // keep the tokens up to date in the event of a refresh
+            oandaBrokerage.SessionRefreshed +=
+                (sender, args) =>
+                {
+                    File.WriteAllText(OandaBrokerageFactory.TokensFile,
+                        JsonConvert.SerializeObject(args, Formatting.Indented));
+                };
+
+            return oandaBrokerage;
+        }
+
+        /// <summary>
+        ///     Gets the symbol to be traded, must be shortable
+        /// </summary>
+        protected override string Symbol
+        {
+            get { return "EUR_USD"; }
```

Comment:
the symbol should be 'EURUSD' like QC expresses it, we'll need to make changes to do the symbol mapping between QC and Oana